### PR TITLE
Calendar: add panel-change event #(16380)

### PR DESCRIPTION
**Pull request by luckyCao**
close feature request #16380

Please make sure these boxes are checked before submitting your PR, thank you!

* [ ] Make sure you follow Element's contributing guide ([中文](https://github.com/ElemeFE/element/blob/master/.github/CONTRIBUTING.zh-CN.md) | [English](https://github.com/ElemeFE/element/blob/master/.github/CONTRIBUTING.en-US.md) | [Español](https://github.com/ElemeFE/element/blob/master/.github/CONTRIBUTING.es.md) | [Français](https://github.com/ElemeFE/element/blob/master/.github/CONTRIBUTING.fr-FR.md)).
* [ ] Make sure you are merging your commits to `dev` branch.
* [ ] Add some descriptions and refer relative issues for you PR.

### CODE REVIEW

File: examples/docs/zh-CN/calendar.md [hunk 1]
```diff
@@ -64,3 +88,8 @@
 |-----------------|-------------- |---------- |------------ |-------- |
 | date            | 单元格代表的日期 | Date      | —           | —      |
 | data            | { type, isSelected, day}，`type` 表示该日期的所属月份，可选值有 prev-month，current-month，next-month；`isSelected` 标明该日期是否被选中；`day` 是格式化的日期，格式为 yyyy-MM-dd    | Object      | —           | —      |
+
+### Calendar Events
+| 事件名 | 说明 | 参数 |
+| ---- | ---- | ---- |
+| panelChange | 点击下个月、下个月或者选择上个月的日期，导致 calendar 面板变化时触发的事件 | pickedDate |
```
Comment (island205) [inline examples/docs/zh-CN/calendar.md]: typo: `点击下个月、下个月`
panelChange 事件名没有直接表达出事件本身，重新考虑一下事件名

Comment (luckyCao) [reply examples/docs/zh-CN/calendar.md]: 我们现在是只有 month 的改变，但是随着功能的迭代，可能有 year 的变化（用户选择下一年，上一年），切换年月的展示等，考虑到上面可能的功能迭代情况，所以用的 panelChange

**Comment by ffey** [comment]
onMonthChange比较合理

File: examples/docs/zh-CN/calendar.md (hunk 1 above)
Comment (ffey) [reply examples/docs/zh-CN/calendar.md]: 应该在点击了右上角的上一月下一月的时候返回当前的月份, 用户可能会根据事件执行相应的操作。

**Comment by hamedbaatour** [comment]
@island205  any plans to merge this pull request? the calendar component is basically unusable now as it doesn't have any event built into it... 😕

**Comment by hamedbaatour** [comment]
@dulinxi @island205 @ziyoung 
any updates on this? 有任何更新吗?

**Comment by antonreshetov** [comment]
any news about this?

**Pull request closed without merging**